File: src/main/java/me/piperswe/demogui/DemoApp.java
package me.piperswe.demogui;

import me.piperswe.shell.StageCommand;
import me.piperswe.utils.gui.ReactiveApplication;

public class DemoApp extends ReactiveApplication<State> {
    public DemoApp() {
        super(State.initialState(), RootComponent::new);
    }

    public static StageCommand getCommand() {
        return new StageCommand(DemoApp::new);
    }

    public static void main(String[] args) {
        launch(DemoApp.class);
    }
}

File: src/main/kotlin/me/piperswe/demogui/kotlin/DemoApp.kt
package me.piperswe.demogui.kotlin

import me.piperswe.shell.StageCommand
import me.piperswe.utils.gui.ReactiveApplication

fun createDemoApplication() = ReactiveApplication(initialState, rootFactory)
val demoApplicationCommand = StageCommand { createDemoApplication() }

fun main() {
    createDemoApplication().launch()
}